src/main/java/org/alliancegenome/curation_api/dao/AffectedGenomicModelDAO.java

```diff
@@ -21,24 +22,33 @@ protected AffectedGenomicModelDAO() {
 		super(AffectedGenomicModel.class);
 	}
 	
-	public List<Long> findReferencingDiseaseAnnotations(Long agmId) {
+	public Boolean hasReferencingDiseaseAnnotations(Long agmId) {
 		Map<String, Object> params = new HashMap<>();
 		params.put("diseaseAnnotationSubject.id", agmId);
 		List<Long> results = agmDiseaseAnnotationDAO.findIdsByParams(params);
+		if (CollectionUtils.isNotEmpty(results)) {
+			return true;
+		}
 		
 		Map<String, Object> dgmParams = new HashMap<>();
 		dgmParams.put("diseaseGeneticModifiers.id", agmId);
-		results.addAll(diseaseAnnotationDAO.findIdsByParams(dgmParams));
+		results = diseaseAnnotationDAO.findIdsByParams(dgmParams);
+		if (CollectionUtils.isNotEmpty(results)) {
+			return true;
+		}
 
-		return results;
+		return false;
 	}
 	
-	public List<Long> findReferencingPhenotypeAnnotations(Long agmId) {
+	public Boolean hasReferencingPhenotypeAnnotations(Long agmId) {
 		Map<String, Object> params = new HashMap<>();
 		params.put("phenotypeAnnotationSubject.id", agmId);
 		List<Long> results = agmPhenotypeAnnotationDAO.findIdsByParams(params);
+		if (CollectionUtils.isNotEmpty(results)) {
```

Comment:
Instead of having the extra condition you could just `return CollectionUtils.isNotEmpty(results);`

Reply:
Yep, fixed here and in other DAOs